### ISSUE TITLE
Nullability warning (CS8602) is triggered despite previous null checks on tested value

### ISSUE THREAD

**Issue by paulosergiojps** [nunit/nunit.analyzers#810]
**Version:** nunit.analyzers 4.4.0  
**.NET Version:** .NET 8

**Code Example:**
```
private static string? GetString()
{
    return "Test";
}

[Test]
public void Test()
{
    string? text = GetString();
    Assert.That(text, Is.Not.Null);
    bool result = text.Contains("other"); // CS8602 warning appears here
    Assert.That(result, Is.True);
}

```

**Expected Behavior:** No nullability warning should be triggered after a non-null assertion, as `text` should be considered proven non-null within that scope.

**Actual Behavior:** A CS8602 warning appears, indicating that the analyzer does not recognize that `Assert.That(text, Is.Not.Null)` ensures `text` is non-null, causing extraneous nullability warnings.

**Comment by manfred-brands** [comment]
Is this happening when you run `dotnet build` or only in Visual Studio? There is a bug in VS 2022 17.12.0. it was fixed in version 17.12.3.

**Comment by manfred-brands** [comment]
See #799

**Comment by manfred-brands** [comment]
I found a [ticket](https://github.com/dotnet/vscode-csharp/issues/7821) in the vs-code repo.
When I switch to a pre-release version of the C# extension, it works for me.

Released version (2.55.29)
![{940C6EE8-5ABC-4A40-AEC7-21920ED82C30}](https://github.com/user-attachments/assets/63a56411-fb3d-40eb-945c-cd1aae0e7b36)

Pre-Release version (2.60.26)
![{26C00B01-76EC-46F7-981E-DDEF47148AB1}](https://github.com/user-attachments/assets/2831602e-5dba-47fc-9a71-78409a9ae419)

As you can see the ones that should be suppressed are and the one that shouldn't is still there (the one inside the Assert.Multiple)

**Comment by paulosergiojps** [comment]
Thank you for your suggestion. It worked well!
